### ISSUE TITLE
[Implementation] Train a seeded logistic-regression baseline

### ISSUE THREAD

**Issue by anilreddy89** [anilreddy89/Inforsight#24]
### Outcome

Implement and document a deterministic logistic-regression benchmark using the frozen Phase 2 observation records, temporal splits, feature dictionary, and training-only preprocessing pipeline.

The completed increment must:

- Fit one versioned logistic-regression specification on the frozen training partition only.
- Produce reproducible train and validation predictions, metrics, coefficients, and artifact digests.
- Keep the canonical test partition sealed.
- Publish a machine-readable experiment manifest and a human-readable experiment report.
- Retain `LIM-002-001` and label all results as pipeline-engineering evidence only.

### Context

This issue implements the next unchecked Phase 2 backlog item:

> Train and document a seeded logistic-regression baseline as the transparent benchmark.

Phase 2.04 established the frozen feature dictionary and preprocessing pipeline in issue #22 and PR #23. The baseline must consume those model matrices without regenerating splits or refitting preprocessing.

The canonical corpus has a claim-blocking limitation: billing frequency is confounded with observation time. Train contains monthly policies, validation contains semiannual policies, and test contains annual policies. Consequently, this issue may demonstrate deterministic model-training mechanics but must not claim temporal generalization, production accuracy, actuarial validity, fairness, or business value.

The implementation plan is documented in:

`Documents/phase-02-05-seeded-logistic-regression-baseline.md`

Relevant repository contracts:

- `docs/modeling/phase-02-01-modeling-contract.md`
- `docs/modeling/phase-02-02-leakage-and-shortcut-guards.md`
- `docs/modeling/phase-02-03-temporal-split-contract.md`
- `docs/modeling/phase-02-04-feature-pipeline-contract.md`
- `docs/modeling/phase-02-04-feature-dictionary.json`
- `docs/limitations.md` (`LIM-002-001`)

### Acceptance checks

- [ ] A versioned baseline-training contract freezes the estimator, solver, penalty, regularization, class-weight, seed, convergence, metric, and artifact rules before results are inspected.
- [ ] The selected modeling dependency is version-constrained and documented.
- [ ] The model consumes the exact frozen Phase 2.04 training matrix.
- [ ] Fitting cannot use validation or test rows or refit preprocessing.
- [ ] Validation is used only for predeclared descriptive diagnostics.
- [ ] The canonical test partition remains sealed and no test metrics or prediction digests are published.
- [ ] Repeated runs reproduce coefficients, intercept, probabilities, metrics, prediction digests, and canonical manifest bytes according to documented reproducibility rules.
- [ ] Convergence warnings and incompatible class ordering, feature ordering, row identities, versions, digests, or non-finite values fail closed.
- [ ] Coefficients and derived odds ratios align one-to-one with the frozen feature names.
- [ ] Coefficient reporting includes non-causal interpretation warnings.
- [ ] Held-out mutations cannot alter fitted model state, while relevant training mutations do alter it.
- [ ] The published manifest records configuration, provenance, fitted parameters, diagnostics, compatibility information, and limitations without raw or transformed row-level matrices.
- [ ] Reloading or reconstructing the published artifact reproduces train and validation predictions without refitting.
- [ ] `LIM-002-001` remains claim-blocking and all metrics are labeled `pipeline_engineering_only`.
- [ ] No boosted model, calibration, operational threshold selection, test-guided choice, or release decision is introduced.
- [ ] Focused baseline tests pass.
- [ ] Existing feature-pipeline, temporal-split, leakage, observation, contract, dataset, and boundary checks continue to pass.
- [ ] Relevant technical and experiment documentation is updated.
- [ ] `make check` and `git diff --check` pass.

### Evidence

Completion should be demonstrated through:

- `docs/modeling/phase-02-05-logistic-baseline-contract.md`
- `docs/experiments/phase-02-05-logistic-baseline-manifest.json`
- `docs/experiments/phase-02-05-logistic-baseline-report.md`
- Focused tests in `simulator/tests/test_logistic_baseline.py`
- Deterministic artifact regeneration through:
  - `python3 scripts/train_logistic_baseline.py --write`
  - `python3 scripts/train_logistic_baseline.py --check`
- Proof that the fitted artifact reproduces train and validation prediction digests without refitting.
- Proof that only the frozen training observation IDs were used during fitting.
- Confirmation that canonical test scoring did not occur.
- Successful output from:
  - `python3 -m unittest discover -s simulator/tests -p 'test_logistic_baseline.py' -v`
  - `python3 scripts/build_feature_pipeline.py --check`
  - `python3 -m unittest discover -s simulator/tests -v`
  - `make check`
  - `git diff --check`
- Pull-request review and passing CI.

### Out of scope

- LightGBM, XGBoost, or another boosted-model candidate.
- Hyperparameter, solver, penalty, or regularization search.
- Validation-driven changes to the frozen baseline specification.
- Resampling, synthetic minority generation, or post-result class weighting.
- Learned feature selection or changes to Phase 2.04 preprocessing.
- Canonical test scoring or test-guided decisions.
- Probability calibration.
- Operational thresholds, review-capacity analysis, high-risk bands, or false-positive cost assumptions.
- SHAP or other local explanation tooling beyond transparent coefficient reporting.
- Resolving `LIM-002-001`.
- Model-card publication, final model selection, release approval, deployment, or serving.
- Real-world performance, temporal-generalization, fairness, actuarial, financial, or customer-impact claims.
- Automated conservation actions or policyholder contact.

### Dependencies

Completed prerequisites:

- Phase 2.01 modeling contract and observation records: issue #16 / PR #17
- Phase 2.02 leakage and simulator-shortcut guards: issue #18 / PR #19
- Phase 2.03 policy-aware temporal splits: issue #20 / PR #21
- Phase 2.04 feature pipeline and training-only preprocessing: issue #22 / PR #23

Active limitation:

- `LIM-002-001` in `docs/limitations.md`

Follow-up:

- Train a frozen boosted-tree candidate and compare it with this benchmark under an approved validation protocol.

### Boundaries

- [x] This issue uses only fictional data, public sources, and original implementation.
- [x] This issue does not include credentials, customer information, or proprietary material.
- [x] This issue preserves point-in-time, authority, and human-review boundaries where applicable.
